lua/iswap/internal.lua

```diff
@@ -47,44 +47,54 @@ function M.get_list_node_at_cursor(winid, config)
     local start_row, start_col, end_row, end_col = node:range()
     local start = { start_row, start_col }
     local end_ = { end_row, end_col }
-    if util.within(start, cursor_range, end_) and node:named_child_count() > 0 then
-      ret = node
+    if util.within(start, cursor_range, end_) and node:named_child_count() > 1 then
+      local children = ts_utils.get_named_children(node)
+      if find_cur_node then
+        local cur_nodes = util.nodes_containing_cursor(children, winid)
+        if #cur_nodes >= 1 then
+          if #cur_nodes > 1 then
+            err("multiple found, using first", config.debug)
+          end
+          ret = { node, children, cur_nodes[1] }
+        end
+      else
+        ret = { node, children }
+      end
     end
   end
   err('completed', config.debug)
-  return ret
+  if ret then
+    return unpack(ret)
+  end
 end
 
-local function private_ts_utils_get_node_text(node, bufnr)
+local function node_or_range_get_text(node_or_range, bufnr)
```

Comment:
why/what did you change about this function?

Reply:
It started with thinking I could just use `vim.treesitter.get_node_text` but that annoyingly doesn't accept a range (which was necessary to support doing multiple swaps for imove, as after the first swap I'm pretty sure the nodes are no longer valid). Looking at get_node_text code in neovims source code it looks cleaner since it uses `nvim_buf_get_text` instead of lines

Reply:
got it